int32 alien_type
float32 centroid_x
float32 centroid_y
